job/service.go

```diff
@@ -170,6 +171,43 @@ func (srv *Service) GetByName(ctx context.Context, name string, namespace models
 	return jobSpec, nil
 }
 
+func (srv *Service) GetByFilter(ctx context.Context, filter models.JobSpecFilter) ([]models.JobSpec, error) {
+	if filter.ResourceDestination != "" {
+		jobSpec, err := srv.interProjectJobSpecRepository.GetJobByResourceDestination(ctx, filter.ResourceDestination)
+		if err != nil {
+			return []models.JobSpec{}, err
```

Comment:
Can you change it to return `nil, err` instead? I believe the slice won't be needed if there's an error. This is to reduce unnecessary memory allocation.

Reply:
cc: @arinda-arif